hack/jenkins/test-integration-dockerized.sh

```diff
@@ -36,7 +36,6 @@ export KUBE_COVER="n"
 export ARTIFACTS=${ARTIFACTS:-"${WORKSPACE}/artifacts"}
 # Save the verbose stdout as well.
 export KUBE_KEEP_VERBOSE_TEST_OUTPUT=y
-export KUBE_INTEGRATION_TEST_MAX_CONCURRENCY=4
```

Comment:
hack/make-rules/test-integration.sh already calls `kube::golang::setup_gomaxprocs`

setting this causes that script to override GOMAXPROCS with KUBE_INTEGRATION_TEST_MAX_CONCURRENCY